src/Atoms/UserProfileData/UserProfileData.js

```diff
@@ -0,0 +1,82 @@
+import propTypes from 'prop-types';
+import { useContext, useEffect, useState } from 'react';
+
+import UserProfileDataStyled from './UserProfileDataStyled';
+import participants from '../../Data/MongoDBMock/participants';
+import { UserDataContext } from '../../Context/UserDataContext';
+
+// import { useLocation } from 'react-router-dom';
+
+const initialData = {
+  address: {
+    country: 'Polska',
+    city: 'Sfornegacie',
+    street: 'ul.Pszczelna',
+    numberOfHouse: '27/5',
+    postalCode: '50-124',
+  },
+};
+
+const UserProfileData = () => {
+  const { state } = useContext(UserDataContext);
+  const { userId, userName, userSurname, isAuthenticated } = state;
+  const [userObject, setUserObject] = useState(initialData);
+
+  if (!isAuthenticated) {
+    throw new Error('Your not allowed to be here!');
+  }
+
+  // mock for checking authentication and if userId is in database
+
+  // const { pathname } = useLocation();
+
+  // if (!isAuthenticated) {
+  //   try {
+  //     const userId = pathname.split('/').pop();
+
+  //     const userObject = participants.find(
+  //       (participant) => participant.participantId === userId,
+  //     );
+  //     dispatch({ type: 'LOG_IN' });
+  //     setUserObject(userObject);
+  //   } catch (err) {
+  //     throw new Error('Your not allowed to be here!');
+  //   }
+  // }
+
+  const { address } = userObject;
+  const { street, numberOfHouse, city, postalCode } = address;
```

Comment:
Theoretically - and in the long run - when this page load, the user object won't exist, so you won't be able to destructure it just yet. The flow will be -> empty state -> fetch -> setState. Only then, you could try to do something with the userObject. For now its ok, as we are basing on mock, but this will have to be reafctored.